### PR TITLE
Add Geometry#typeName method [publish docs]

### LINKED ISSUE #670: Expose a means of converting Geometry type integers to strings
The method http://mapnik.org/node-mapnik/documentation/3.5/#Geometry.type seems to be returning an integer, 0-7, which I then have to convert to a string: https://github.com/mapbox/mapbox-geostats/blob/master/lib/type-integer-to-string.js.

It would be nice if node-mapnik made this easier by either:

- exposing a conversion function,
- exposing an object mapping numbers to strings, or
- returning the string geometry name from `geometry.type`.

What do you think?

### PR DESCRIPTION
**Pull request by davidtheclark**
Closes #670.

cc @springmeyer.

I was unable to get the documentation generated locally (Am I missing something? Is there a particular `documentation` version you need globally installed?), but tried adding a docblock comment to the relevant `*.cpp` file.

### CODE REVIEW

**Comment by springmeyer** [comment]
Looks great. As far as documentation - our system is only set up (I think) to document the cpp files. Interesting challenge trying to document both. /cc @mapsam for ideas.

**Comment by mapsam** [comment]
@davidtheclark @springmeyer interesting, my first thought is that documentationjs is not going to work in `.js` files when `--polyglot` mode is in action. We'd have to do some sort of combination post-generation. I'll test it out.

Regardless, the `[publish docs]` commit message is busted - sorry. I'll fix that!

**Comment by davidtheclark** [comment]
@mapsam @springmeyer Any update on this? Anything you'd like me to change or do to help get this merged?

**Comment by mapsam** [comment]
@davidtheclark re: docs - they've been fixed (we're actually no longer building via commit messages, but no need to update here).

screen shot 2016-09-02 at 10 57 05 am

Looking good - I fixed the package.json so you can run docs locally `npm run docs`. It's a different theme, but the information checks out.

File: lib/mapnik.js [hunk 1]
```diff
@@ -86,6 +86,33 @@ if (process.env.MAPNIK_FONT_PATH) {
...
+    case mapnik.Geometry.Unknown:
+      return 'Unknown';
+    default:
+      return 'Unknown';
+  }
+}
+
+mapnik.Geometry.prototype.typeName = function() {
+  return getGeometryType(this.type());
+}
```
Comment (mapsam) [inline lib/mapnik.js]: Linter is failing here with the missing semicolon @davidtheclark

**Comment by davidtheclark** [comment]
@mapsam Agh. Fixed the linting issue. The other test failures seem unrelated — not sure if you want me to look into those?

**Comment by mapsam** [comment]
Thanks @davidtheclark! The appveyor failures are definitely unrelated. Looks like TravisCI is passing nicely now - I'll wait for it to finish up.

Ah, could you add a CHANGELOG update as well? When you commit, add the following to prevent another build `[skip ci]`

**Comment by mapsam** [comment]
Tests passing 👍 from me - @springmeyer look good to you?

**Comment by springmeyer** [comment]
👍  merge-away. Nice work on this! Thank you.

**Comment by davidtheclark** [comment]
Possible changelog entry:

> Added `typeName` to `mapnik.Geometry`. This returns the GeoJSON type name of a geometry.

**Pull request merged**